### ISSUE TITLE
hub not respecting my $PATH when opening editor?

### ISSUE THREAD

**Issue by bobwhitelock** [github/hub#1675]
Hi, apologies if this is already a known issue, I searched and couldn't find anything which seemed exactly the same, although https://github.com/github/hub/issues/1482 and https://github.com/github/hub/issues/721 look like they might be related.

Basically I have `$EDITOR` set to `vim`, and a couple of versions of `vim` installed on my system:

```zsh
$ echo $EDITOR
vim
$ which vim
/usr/local/bin/vim
$ where vim
/usr/local/bin/vim
/usr/bin/vim
/usr/local/bin/vim
/usr/local/bin/vim
/usr/local/bin/vim
/usr/local/bin/vim
$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Sep 18 2016 15:04:51)
Included patches: 1-5
[... snip ...]
$ /usr/bin/vim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Mar 10 2017 00:51:06)
Included patches: 1-488, 576
Extra patches: 8.0.0378, 8.0.0377, 8.0.0322, 8.0.0056
```

The former is much newer, appears first in `$PATH`, and is the one that is used by `git` normally because of this:
```zsh
$ command git commit
:version                                                                                                                                                                                                                                                                          
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Sep 18 2016 15:04:51)                                                                                                                                                                                                                
Included patches: 1-5
[... snip ...]
```

However when I use `hub` this is not picked up and the other version is used:
```zsh
$ hub commit
YouCompleteMe unavailable: requires Vim 7.4.1578+.
vim-go requires Vim 7.4.1689 or Neovim, but you're using an older version.
Please update your Vim for the best vim-go experience.
If you really want to continue you can set this to make the error go away:
    let g:go_version_warning = 0
Note that some features may error out or behave incorrectly.
Please do not report bugs unless you're using Vim 7.4.1689 or newer.
Press ENTER or type command to continue
```

I did not previously have `$GIT_EDITOR` or `core.editor` in my `~/.gitconfig` set to anything, as just having `git` use `$EDITOR` is the behaviour I wanted. However by explicitly setting `$GIT_EDITOR` to the editor I want to use I can prevent this issue:
```zsh
export GIT_EDITOR=/usr/local/bin/vim
```

So although it's not ideal that `hub` changes/breaks `git`s behaviour for this, I can work around it. Thanks

**Comment by mislav** [comment]
How did you install hub? Which version?

`hub commit` just proxies to `git commit` without any changes to PATH or environment variables. It's strange that it would behave differently. More information would be useful. Thank you!

**Comment by bobwhitelock** [comment]
Hi - thanks for your response. That is odd that this behaviour would occur. I'm using the latest non-pre-release version of `hub`:

```bash
$ hub --version
git version 2.1.4
hub version 2.2.9
```
which I installed by downloading `hub-linux-amd64-2.2.9.tgz` from the releases page, extracting this, and moving the `hub` binary to `/usr/local/bin/hub`.

I've tried using a fresh terminal without my `~/.zshrc` etc loaded to see if anything from my environment could be causing this, but I still get the same behaviour. In case this is helpful, in this case without any shell setup files loaded my `PATH` is just ` /usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games`.

**Comment by bobwhitelock** [comment]
For what it's worth, I've just tried downloading and installing `hub 2.3.0-pre10` and I get the same behaviour if I don't explicitly set `GIT_EDITOR` to avoid this:
```bash
$ hub --version
git version 2.1.4
hub version 2.3.0-pre10
$ hub commit
YouCompleteMe unavailable: requires Vim 7.4.1578+.
vim-go requires Vim 7.4.1689 or Neovim, but you're using an older version.
Please update your Vim for the best vim-go experience.
If you really want to continue you can set this to make the error go away:
    let g:go_version_warning = 0
Note that some features may error out or behave incorrectly.
Please do not report bugs unless you're using Vim 7.4.1689 or newer.
Press ENTER or type command to continue
```

**Comment by mislav** [comment]
@bobwhitelock Thank you for the additional info. What is the output of your `git var GIT_EDITOR`?

**Comment by bobwhitelock** [comment]
Without explicitly setting `$GIT_EDITOR` to the correct path as above (in which case `git var GIT_EDITOR` just gives that path), I get:
```bash
$ command git var GIT_EDITOR
vim
$ hub var GIT_EDITOR    
vim
```

**Comment by bobwhitelock** [comment]
Hi again - I recently switched to a new computer where I'm now using Git version 2.17.0, and this issue no longer seems to occur. I'm guessing whatever was causing this was a Git issue, which must have been fixed sometime between 2.1.4 and 2.17.0, so I'll close this.

**Comment by bobwhitelock** [comment]
Thinking again about what my original issue was, it's possible this only stopped occurring as I no longer have multiple different Vims installed so hub could no longer pick the incorrect one. In any case, I'll leave it closed since either way it's no longer effecting me.
